clients/src/main/java/org/apache/kafka/common/security/authenticator/AuthCallbackHandler.java

```diff
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.kafka.common.security.auth;
+package org.apache.kafka.common.security.authenticator;
```

Comment:
Do we need an upgrade note for this and the other package change? It's a bit odd because they were sharing a package with a public class although we didn't generate javadocs for it.

Reply:
A note wouldn't hurt. I'll add one.